engine/internal/runtime/client.go

```diff
@@ -137,36 +138,60 @@ func (c *commonClient) deployRuntime(
 
 	resConf := c.getResouces(params.modelID)
 
-	sharedVolume := corev1apply.Volume().WithName(shareVolName)
-	var forcePull bool
-	if resConf.Volume != nil {
-		sharedVolume.WithPersistentVolumeClaim(
-			corev1apply.PersistentVolumeClaimVolumeSource().
-				WithClaimName(name))
-	} else {
-		sharedVolume.WithEmptyDir(corev1apply.EmptyDirVolumeSource())
-		// Make every pod pull the model on startup as they don't share the volume.
-		forcePull = true
-	}
 	volumes := []*corev1apply.VolumeApplyConfiguration{
-		sharedVolume,
 		corev1apply.Volume().
 			WithName(configVolName).
 			WithConfigMap(corev1apply.ConfigMapVolumeSource().
 				WithName(c.rconfig.ConfigMapName)),
 	}
-	volumes = append(volumes, params.volumes...)
+	var volClaim *corev1apply.PersistentVolumeClaimApplyConfiguration
+	var forcePull bool
+	if vol := resConf.Volume; vol != nil {
+		pvcName := volName
+		if resConf.Volume.ShareWithReplicas {
+			pvcName = name
+			volumes = append(volumes, corev1apply.Volume().WithName(volName).
+				WithPersistentVolumeClaim(
+					corev1apply.PersistentVolumeClaimVolumeSource().
+						WithClaimName(name)))
+		} else {
+			forcePull = true
```

Comment:
Could you add a comment here? Couldn't follow why we don't need to append to `volumes`.